samcli/commands/build/build_context.py

```diff
@@ -591,7 +591,7 @@ def _collect_single_function_and_dependent_layers(
             return
 
         resource_collector.add_function(function)
-        resource_collector.add_layers([l for l in function.layers if l.build_method is not None and not l.skip_build])
+        resource_collector.add_layers([l for l in function.layers if BuildContext.is_layer_buildable(l)])
```

Comment:
Interesting. I wonder if there was a specific reason for this. Also looks like the `function` part doesn't filter. I assume that is likely a bug? If so, just want to note it for us to fix later (outside this pr).

Reply:
I am not sure. Existing logic was pretty same with `is_layer_buildable`. Only difference is, `is_layer_buildable` additionally checks if `codeuri` of the layer points to a ZIP file or not.

For function collection part, even if function is set `SkipBuild: True`, then running `sam build {logical-id}` will build the function. But if you run `sam build` to build entire stack, it will skip that function to build. So it feels like this is not the expected behavior.